src/Listeners/RemoveClaimToken.php

```diff
@@ -15,21 +18,41 @@ class RemoveClaimToken implements ShouldQueue
      */
     public function handle(PlatformBroadcastEvent $event): void
     {
-        BeamClaim::where('token_chain_id', $event->broadcastData['tokenId'])
-            ->whereHas(
-                'collection',
-                fn ($query) => $query->where('collection_chain_id', $event->broadcastData['collectionId'])
-            )
-            ->claimable()
-            ->with('beam')
-            ->get()
-            ->each(function ($claim) {
-                if ($claim->token?->nonFungible) {
-                    $claim->delete();
-                    if ($code = $claim?->beam?->code) {
-                        Cache::decrement(BeamService::key($code));
-                    }
+        $collectionId = Cache::remember(
+            'collection:' . $event->broadcastData['collectionId'],
+            30,
+            fn () => Collection::where('collection_chain_id', $event->broadcastData['collectionId'])->value('id')
+        );
+
+        if (!$collectionId) {
+            return;
+        }
+
+        $key = 'RemoveClaimToken:' . $collectionId . ':' . $event->broadcastData['tokenId'];
+        Cache::lock($key)->get(function () use ($event, $collectionId) {
```

Comment:
```suggestion
        Cache::lock($key, 5)->get(function () use ($event, $collectionId) {
```

Let's automatically release the key after 5 seconds, just in case anything happens (such as the server crashes).